### PR TITLE
First post by Devon, on using SNMP to monitor QNAP

### PR DESCRIPTION
**Pull request by zmustapha**

### CODE REVIEW

File: content/articles/qnap-monitoring-environment.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+socmediaimg: "socmediaimg-qnap-monitoring-environment.jpg"
+hiliteimg: "poster-qnap-monitoring-environment.jpg"
+poster_sourceurl: "https://www.pinclipart.com/pindetail/Txhmbi_office-management-clipart-performance-monitoring-png-download/"
+poster_source: "Pinclipart.com"
+bookendanimal: "satellite-dish"
+---
+
+At the CTL, we use Graphite and Grafana to monitor all of our server
+infrastructure, but our last physical server, a QNAP file server, does not have
+a straightforward way to send it’s metrics to a graphite monitoring
```
Comment (sdreher) [inline content/articles/qnap-monitoring-environment.md]: its

File: content/articles/qnap-monitoring-environment.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+poster: "poster-qnap-monitoring-environment.jpg"
+thumbnail: "poster-qnap-monitoring-environment.jpg"
+socmediaimg: "socmediaimg-qnap-monitoring-environment.jpg"
+hiliteimg: "poster-qnap-monitoring-environment.jpg"
+poster_sourceurl: "https://www.pinclipart.com/pindetail/Txhmbi_office-management-clipart-performance-monitoring-png-download/"
+poster_source: "Pinclipart.com"
+bookendanimal: "satellite-dish"
+---
+
+At the CTL, we use Graphite and Grafana to monitor all of our server
```
Comment (nikolas) [inline content/articles/qnap-monitoring-environment.md]: I would like to the Graphite and Grafana tools here.

* https://graphite.readthedocs.io/en/latest/
* https://grafana.com/

File: content/articles/qnap-monitoring-environment.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+infrastructure, but our last physical server, a QNAP file server, does not have
+a straightforward way to send its metrics to a graphite monitoring
+environment.  Even worse, it runs a customized version of Linux which we were
+hesitant to modify.  Luckily it does support the Simple Network Management
+Protocol (SNMP), which can do just that.
+
+## What is SNMP?
+
+Essentially it is a network protocol for collecting information about managed
```
Comment (nikolas) [inline content/articles/qnap-monitoring-environment.md]: "Essentially" seems unnecessary here. I would just say, "It is a network protocol for collecting information about managed devices."

File: content/articles/qnap-monitoring-environment.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+Install collectd:
+
+```
+apt-get install collectd
+```
+
+Modify the configuration file at /etc/collectd/collectd.conf.  The default
+config file has lots of options, which are mostly disabled and commented out.
+I wont focus on the global configuration, but there are a few variables to be
```
Comment (nikolas) [inline content/articles/qnap-monitoring-environment.md]: wont -> won't

File: content/articles/qnap-monitoring-environment.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+set like hostname, logging options, plugin directory, etc.
+
+We’ll be enabling the SNMP and write_graphite plugins.  The
+[SNMP](https://collectd.org/documentation/manpages/collectd-snmp.5.shtml)
+manual page of collectd has detailed information about how to configure the
+plugins, but each metric has to be specified indivdually.  For example, to get
```
Comment (nikolas) [inline content/articles/qnap-monitoring-environment.md]: indivdually -> individually

File: content/articles/qnap-monitoring-environment.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+a straightforward way to send its metrics to a graphite monitoring
+environment.  Even worse, it runs a customized version of Linux which we were
+hesitant to modify.  Luckily it does support the Simple Network Management
+Protocol (SNMP), which can do just that.
+
+## What is SNMP?
+
+Essentially it is a network protocol for collecting information about managed
+devices.  Traditionally it is used by many types of networking equipment,
+including switches, routers, etc.  It’s paired with an MIB (Management
```
Comment (nbuonin) [inline content/articles/qnap-monitoring-environment.md]: `paired with a MIB`, I don't think you need the `n`

File: content/articles/qnap-monitoring-environment.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+apt-get install snmp
+```
+
+We then have to add the MIB file to the snmp watch directory, which by default
+is either `$HOME/.snmp/mibs` or `/usr/local/share/snmp/mibs`.  Running the
+command `net-snmp-config --default-mibdirs` will also give you the MIB
+directory location. 
+
+Console output from this tool is very hard to read, so I used one of the many
+available MIB browsers to see what data can be polled.  There are many out
```
Comment (nbuonin) [inline content/articles/qnap-monitoring-environment.md]: I think `to see which data`

**Pull request merged**